src/html/html-script-tag.ts

```diff
@@ -24,13 +24,11 @@ export class ScriptTagImport extends Import { readonly type = 'html-script'; }
 
 export class ScannedScriptTagImport extends ScannedImport {
   resolve(document: Document): ScriptTagImport|undefined {
-    if (this.url === undefined) {
-      return;
-    }
-    const resolvedUrl = document._analysisContext.resolver.resolve(
-        document.parsedDocument.baseUrl, this.url, this);
-    if (resolvedUrl === undefined) {
-      return;
+    const resolvedUrl = this.getLoadableUrlOrWarn(document);
+    if (this.url === undefined || resolvedUrl === undefined) {
```

Comment:
One wait to avoid this duplication would be to return this.url _and_ resolvedUrl from getLoadableUrlOrWarn